src/index.js

```diff
@@ -248,12 +258,16 @@ function fakeDataForProps (props = {}, { optional = false } = {}) {
 
 module.exports = function (file, { optional = false, all = false } = {}) {
   const source = fs.readFileSync(file)
+  const isTsFile = file.match(/^.*\.(ts|tsx)$/)
+  const options = { filename: isTsFile ? file : undefined }
```

Comment:
We need to provide the filename option here for ts files (https://github.com/reactjs/react-docgen/issues/355#issuecomment-490521336), however if we use this option for flow files react-docgen starts to throw some errors, so I'm only using this for `.ts|.tsx` extensions.